src/components/Todos/Todos.tsx

```diff
@@ -0,0 +1,210 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  deleteTodo,
+  getTodos,
+  postTodo,
+  modifyTodo,
+} from '../../api/todos';
+
+import { AuthContext } from '../Auth/AuthContext';
+
+import { Todo } from '../../types/Todo';
+import { TodoModifier } from '../../types/TodoModifier';
+import { ErrorType } from '../../types/ErrorType';
+import { FilterType } from '../../types/FilterType';
+
+import { Notification } from '../Notification';
+import { TodoAddForm } from '../TodoAddForm';
+import { TodoList } from '../TodoList';
+import { BottomBar } from '../BottomBar';
+import { TodosStatusToggler } from '../TodosStatusToggler';
+
+export const Todos: React.FC = React.memo(() => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(ErrorType.None);
+  const [filter, setFilter] = useState<FilterType>(FilterType.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [modifyingTodosId, setModifyingTodosId] = useState<number[]>([11111]);
+
+  const user = useContext(AuthContext);
+
+  const loadTodos = useCallback(async () => {
+    if (user) {
+      try {
+        setTodos(await getTodos(user.id));
+      } catch {
+        setErrorMessage(ErrorType.Load);
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos().then();
+  }, [user]);
+
+  const visibleTodos = useMemo(() => (
+    todos.filter(todo => {
+      switch (filter) {
+        case FilterType.All:
+          return true;
```

Comment:
return todo;

Reply:
The Filter method needs a callback that returns true or false (predicate). So I'm in return and return true.